File: lib/csharp/microsoft.bot.builder.skills/Microsoft.Bot.Builder.Skills/Models/SkillDefinition.cs
namespace Microsoft.Bot.Builder.Skills
{
    public class SkillDefinition
    {
        public string Name { get; set; }

        public string DispatchIntent { get; set; }

        public string Endpoint { get; set; }    

        public string[] SupportedProviders { get; set; }

        public string Scope { get; set; }
    }
}

File: solutions/Virtual-Assistant/src/typescript/bot-skill/src/Models/skillDefinition.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

export class SkillDefinition {

    public name: string = '';
    public dispatchIntent: string = '';
    public endpoint: string = '';
    public supportedProviders: string[] = [];
    public scope: string = '';
}
